### ISSUE TITLE
Duplicates in favorited locations

### ISSUE THREAD

**Issue by PhilC813** [Akylas/oss-weather#271]
On the Android app, if I use the "Find GPS location button" option in the overflow menu, and then favorite the location, I can repeat these steps and favorite the same location over and over again, leading to duplicates in the favorites menu.

Install source: Google Play
App version: 2.9.3 (latest)
Build: 146

Device: REDMAGIC 9 Pro
Android version: 14 (latest)

**Comment by farfromrefug** [comment]
@PhilC813 dont think i will fix this. The favorite id is defined by its lat long. If you use the gps multiple times it might return different lat/lng. I cant know that those are actually the same. And users might want close locations (weather forecast might be different).
SO dont think i will fix it

**Comment by PhilC813** [comment]
Oh okay I understand, but the UI really gives the impression that it's based on postal code. Most weather apps don't work this way. Could the coordinates be specified in the list instead?

**Comment by farfromrefug** [comment]
@PhilC813 good point Will add it

**Comment by PhilC813** [comment]
Thank you!!

Also, have you seen this UI issue before? The one that you can see in my screenshot, with the top of the favorites sidebar cut off. It only happens with the favorites menu and maybe one or two other areas but can't recall where.

**Comment by farfromrefug** [comment]
@PhilC813 no i have not will try to reproduce to fix it

**Comment by 0-5-0** [comment]
I'm getting a weird issue with the new 147 version, if I open the favorites section, tap on my saved favorite, it then opens it in the main view but I can't scroll down anymore.

If I go back to the three lines menu in the top left and just tap anywhere that isn't a saved favorite then I can tap on the right to go back to the main view and scrolling works again. If that explanation all makes sense..

Basically open favorites, tap your favorite, try to scroll the main view.. go back to favorite, tap the favorite again, try to scroll, go back to favorite again, tap anywhere but the saved favorite, scroll works again.

Seems like a similar bug to when the pop-up settings menus weren't scrollable a couple weeks ago.

**Comment by PhilC813** [comment]
With the new 147 build, I'm now seeing the coordinates!

The issue with the favorites sidebar being cut off at the top of the screen also seems resolved! 

Thank you!

----------

Otherwise, I did try to replicate what @0-5-0 mentioned. To summarize (hope you don't mind), he alleges that you can't scroll down after selecting an entry in the favorites sidebar.

It took me a few tries, switching back and forth between my (only) two favorites, and I did manage to replicate it, albeit only once. So although it may not be replicable every time, there's definitely something going wrong here.

----------

Separately, there is a new small layout issue. As you can see in the screenshot below, the coordinates can get cut off. I've also seen it being cut off for both entries at some point.

![Screenshot_20241218_022234](https://github.com/user-attachments/assets/e25e09e5-0110-489d-8ffd-fce5ec142143)

**Comment by farfromrefug** [comment]
@PhilC813 @0-5-0 thanks to you both for the great explanations! i fixed the cut off text. Also tried to fix the touch issue though i could not reproduce it

**Comment by 0-5-0** [comment]
I removed the existing favorite, re-added it and it all seems fine now, scrolling works.

**Comment by PhilC813** [comment]
@farfromrefug I don't like bearing bad news but the coordinates still get cut off in build 148 😅

![Screenshot_20241218_101437](https://github.com/user-attachments/assets/9ded9eee-b788-4e7c-aeb1-54ee7254b658)

**Comment by farfromrefug** [comment]
@PhilC813 i cant reproduce with latest. Are you still using it? can try removing the favorite and add it again?

**Comment by PhilC813** [comment]
@farfromrefug It looks okay with the latest version, without having to re-add the locations. I'll leave a comment if it occurs again. Thank you!
